File: frontend/src/utils/importParser.ts
export interface ParsedLine {
  email: string;
  password: string;
  client_id: string;
  refresh_token: string;
}

export interface ParseError {
  line_number: number;
  content: string;
  reason: string;
}

export interface ParseResult {
  valid: ParsedLine[];
  errors: ParseError[];
}

export function parseImportText(text: string): ParseResult {
  const valid: ParsedLine[] = [];
  const errors: ParseError[] = [];

  const lines = text.split('\n');
  for (let i = 0; i < lines.length; i++) {
    const line = lines[i].trim();
    if (!line) continue;

    let parts: string[];
    if (line.includes('----')) {
      parts = line.split('----').map(p => p.trim());
    } else {
      parts = line.split(/\s+/);
    }

    if (parts.length !== 4) {
      errors.push({
        line_number: i + 1,
        content: line,
        reason: `字段数量不足（期望 4 个字段，实际 ${parts.length} 个）`,
      });
    } else {
      valid.push({
        email: parts[0],
        password: parts[1],
        client_id: parts[2],
        refresh_token: parts[3],
      });
    }
  }

  return { valid, errors };
}

File: backend/app/services/import_parser.py
from dataclasses import dataclass


@dataclass
class ParsedMailbox:
    email: str
    password: str
    client_id: str
    refresh_token: str


@dataclass
class ImportError:
    line_number: int
    content: str
    reason: str


@dataclass
class ParseResult:
    valid: list[ParsedMailbox]
    errors: list[ImportError]


def _parse_line(line: str) -> ParsedMailbox | None:
    if "----" in line:
        parts = [p.strip() for p in line.split("----")]
    else:
        parts = line.split()

    if len(parts) != 4:
        return None

    return ParsedMailbox(
        email=parts[0].strip(),
        password=parts[1].strip(),
        client_id=parts[2].strip(),
        refresh_token=parts[3].strip(),
    )


def parse_import_text(text: str) -> ParseResult:
    valid: list[ParsedMailbox] = []
    errors: list[ImportError] = []

    for line_num, raw_line in enumerate(text.splitlines(), start=1):
        line = raw_line.strip()
        if not line:
            continue

        parsed = _parse_line(line)
        if parsed is None:
            field_count = len(line.split("----")) if "----" in line else len(line.split())
            errors.append(ImportError(
                line_number=line_num,
                content=line,
                reason=f"字段数量不足（期望 4 个字段，实际 {field_count} 个）",
            ))
        else:
            valid.append(parsed)

    return ParseResult(valid=valid, errors=errors)
